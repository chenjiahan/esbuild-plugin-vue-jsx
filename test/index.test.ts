import path from 'path';
import { build } from 'esbuild';
import vueJsx from '../src';

test('simple', async () => {
  const result = await build({
    absWorkingDir: process.cwd(),
    write: false,
    outdir: './test/fixture/dist',
    bundle: true,
    entryPoints: ['./test/fixtures/foo.jsx', './test/fixtures/bar.tsx'],
    plugins: [vueJsx()],
    external: ['vue'],
    format: 'esm',
  });
  for (const file of result.outputFiles) {
    expect(file.text).toMatchSnapshot(path.relative(process.cwd(), file.path));
  }
});
