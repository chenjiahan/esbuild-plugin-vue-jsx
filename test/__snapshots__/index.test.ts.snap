// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple: test/fixture/dist/bar.js 1`] = `
"// vue-jsx:./test/fixtures/bar.tsx
import { createVNode as _createVNode, createTextVNode as _createTextVNode } from \\"vue\\";
var bar_default = {
  render() {
    return _createVNode(\\"div\\", null, [_createTextVNode(\\"Hello world\\")]);
  }
};
export {
  bar_default as default
};
"
`;

exports[`simple: test/fixture/dist/foo.js 1`] = `
"// vue-jsx:./test/fixtures/foo.jsx
import { createVNode as _createVNode, createTextVNode as _createTextVNode } from \\"vue\\";
var foo_default = {
  render() {
    return _createVNode(\\"div\\", null, [_createTextVNode(\\"Hello world\\")]);
  }
};
export {
  foo_default as default
};
"
`;
